### ISSUE TITLE
Wrapping onnx structs

### ISSUE THREAD

**Issue by nopeslide** [alrevuelta/cONNXr#39]
Instead of using onnx structs directly I would like to wrap them in a simple manner like:

```c
typedef struct {
   Onnx__TensorProto onnx;
} tensor;

So if we need to extend any structure given by onnx we can do it inside our own wrapper without having to refactor everything.
We essentially did this already with the node_context in regard to Onnx__NodeProto.
```

**Comment by nopeslide** [comment]
One thing I have in mind is to add a `void` pointer to all our (onnx) structures which can be used when `cONNXr` is integrated in some other codebase to easily extend it without replacing internal allocation code

**Comment by alrevuelta** [comment]
Right, this is essentially what we did with `node_context` and `Onnx__NodeProto`. Its a fair suggestion but tbh I don't really foresee any extension, so I would leave as it is by now and focus our efforts somewhere else. If in the future we find the need, we can of course wrap it.

**Comment by nopeslide** [comment]
I would like to do it in near future, since I need to integrate connxr in another framework and therefore need to replace/extend a few things.

**Comment by alrevuelta** [comment]
If you have already thought a particular use case I'm fine with it.
